lib/fog/compute/kubevirt.rb

```diff
@@ -84,26 +118,142 @@ def initialize(options={})
           # Kubeclient needs different client objects for different API groups. We will keep in this hash the
           # client objects, indexed by API path/version.
           @clients = {}
+        end
 
-          @client = kubevirt_client()
+        def virt_supported?
+          virt_enabled = false
 
-          # TODO expect a specific token
-          @oc_client = openshift_client()
-          @kube_client = kube_client()
+          begin
+            api_versions = kubevirt_client.api["versions"]
+            api_versions.each do |ver|
+              if ver["groupVersion"]&.start_with?(KUBEVIRT_GROUP)
+                virt_enabled = true
+              end
+            end
+          rescue => err
+            # we failed to communicate or to evaluate the version format
+            $log.warn("Failed to detect kubevirt on provider with error: #{err.message}")
+          end
+
+          virt_enabled
         end
 
-        private
+        def valid?
+          kube_client.api_valid?
+        end
 
-        def client
-          @client
+        class WatchWrapper
```

Comment:
We should add `finish` method to the `Wrapper` and delegate the call to `@watch` so we can stop the watches properly.